### DISCUSSION TITLE
LS82-Tuya curtain motors, show up, return correct position but do not move when the service is called without an error message.

### DISCUSSION THREAD

**Discussion by DavoudTaghawiNejad** [zigpy/zha-device-handlers#2282] [General]
Hello

I am trying to add a quirk for  the "LS82-Tuya" curtain motor (cover). This is a zigbee device that looks exactly as other tuya curtain motors. I was able to build the signature, so that it is recognized. It shows up as a cover device. And it correctly reflects the position (open/closed). However, when I push the open stop or close button nothing happens. There is also no error message in the logs. 
I am at loss how to trouble shoot this device. The current quirk is:

    class LS82TuyaCover(TuyaWindowCover):
        """Tuya blind controller device."""
        signature = {
            # "node_descriptor": "NodeDescriptor(byte1=2, byte2=64, mac_capability_flags=128, manufacturer_code=4098,
            #                    maximum_buffer_size=82, maximum_incoming_transfer_size=82, server_mask=11264,
            #                    maximum_outgoing_transfer_size=82, descriptor_capability_field=0)",
            # "endpoints": {
            # "1": { "profile_id": 260, "device_type": "0x0051", "in_clusters": [ "0x0000", "0x0004","0x0005","0xef00"], "out_clusters": ["0x000a","0x0019"] }
            # },
            # "manufacturer": "_TZE200_xu4a5rhj",
            # "model": "TS0601",
            # "class": "zigpy.device.Device"
            MODELS_INFO: [("_TZE200_xu4a5rhj", "TS0601")],
            ENDPOINTS: {
                1: {
                    PROFILE_ID: zha.PROFILE_ID,
                    DEVICE_TYPE: zha.DeviceType.SMART_PLUG,
                    INPUT_CLUSTERS: [
                        Basic.cluster_id,
                        Groups.cluster_id,
                        Scenes.cluster_id,
                        TuyaManufCluster.cluster_id,
                    ],
                    OUTPUT_CLUSTERS: [Time.cluster_id, Ota.cluster_id],
                },
                242: {
                    PROFILE_ID: 41440,
                    DEVICE_TYPE: 0x0061,
                    INPUT_CLUSTERS: [],
                    OUTPUT_CLUSTERS: [0x0021
                        ]
                }
            },
        }
    
        replacement = {
            ENDPOINTS: {
                1: {
                    DEVICE_TYPE: zha.DeviceType.WINDOW_COVERING_DEVICE,
                    INPUT_CLUSTERS: [
                        Basic.cluster_id,
                        Groups.cluster_id,
                        Scenes.cluster_id,
                        TuyaManufacturerWindowCover,
                        TuyaWindowCoverControl,
                    ],
                    OUTPUT_CLUSTERS: [Time.cluster_id, Ota.cluster_id],
                },
                242: {
                    DEVICE_TYPE: 0x0061,
                    INPUT_CLUSTERS: [],
                    OUTPUT_CLUSTERS: [0x0021
                        ]
                }
            }
        }

**Comment by DavoudTaghawiNejad** [comment]
The signature is:
    
    {
      "node_descriptor": "NodeDescriptor(logical_type=<LogicalType.Router: 1>, complex_descriptor_available=0, user_descriptor_available=0, reserved=0, aps_flags=0, frequency_band=<FrequencyBand.Freq2400MHz: 8>, mac_capability_flags=<MACCapabilityFlags.AllocateAddress|RxOnWhenIdle|MainsPowered|FullFunctionDevice: 142>, manufacturer_code=4417, maximum_buffer_size=66, maximum_incoming_transfer_size=66, server_mask=10752, maximum_outgoing_transfer_size=66, descriptor_capability_field=<DescriptorCapability.NONE: 0>, *allocate_address=True, *is_alternate_pan_coordinator=False, *is_coordinator=False, *is_end_device=False, *is_full_function_device=True, *is_mains_powered=True, *is_receiver_on_when_idle=True, *is_router=True, *is_security_capable=False)",
      "endpoints": {
        "1": {
          "profile_id": 260,
          "device_type": "0x0202",
          "in_clusters": [
            "0x0000",
            "0x0004",
            "0x0005",
            "0x0102",
            "0xef00"
          ],
          "out_clusters": [
            "0x000a",
            "0x0019"
          ]
        },
        "242": {
          "profile_id": 41440,
          "device_type": "0x0061",
          "in_clusters": [],
          "out_clusters": [
            "0x0021"
          ]
        }
      },
      "manufacturer": "_TZE200_xu4a5rhj",
      "model": "TS0601",
      "class": "ts0601_cover.UnknownCover"
    }

**Comment by Hedda** [comment]
@DavoudTaghawiNejad Open a new issue with "Device support request" -> https://github.com/zigpy/zha-device-handlers/issues

There you provide the requested device signature, diagnostic information, logs, etc. which makes it easier for others to answer and help.

PS: This is in ZHA docs -> https://www.home-assistant.io/integrations/zha#how-to-add-support-for-new-and-unsupported-devices
